dcos/http.py

```diff
@@ -93,7 +100,7 @@ def _verify_ssl(url, verify=None, toml_config=None):
 def _request(method,
              url,
              is_success=_default_is_success,
-             timeout=DEFAULT_TIMEOUT,
+             timeout=-1,
```

Comment:
Please document this in the docstring.

Reply:
This `-1` system is not expected behaviour as a user of this API.

What do you think of instead making `timeout` a required parameter? Perhaps there could be a helper `http_timeout_from_configuration` and some callers would use that.

Reply:
Looking closer at the underlying library (requests), their timeout can actually be splitted in a tuple : 
```
 :param timeout: (optional) How many seconds to wait for the server to send data
                            before giving up, as a float, or a :ref:`(connect timeout, read
                            timeout) <timeouts>` tuple.
```

I think we need to leverage this, the `FAIL_FAST_TIMEOUT` would become the connection timeout, `DEFAULT_TIMEOUT` would become the timeout for reading the response. Then I am inclined to bump the default timeout way more (even to the documented 180 seconds).

The timeout being a required parameter would make the http module way less user-friendly. I agree though that `-1` seems odd. We need to have the default being read from configuration or falling back to the constant, I can suggest instead of `-1` to use a default of `True`, which makes more sense imo.

WDYT?